### PR TITLE
Add security documentation

### PR DESCRIPTION
**Pull request by OR13**
Ignore build server, its just reminding us most did methods are not implementing the did core context correctly.

### CODE REVIEW

**Comment by OR13** [comment]
Almost all of security-v1, security-v2, and did-core are covered with at least links to relevant ccg specs.

File: security.md [hunk 1]
```diff
@@ -10,51 +10,61 @@ Deprecated signature suite, do not use.
...
 
-Deprecated, do not use.
+The cipher algorithm describes the mechanism used to encrypt a message. It is typically a string expressing the cipher suite, the strength of the cipher, and a block cipher mode.
+
+See [w3c-ccg](https://w3c-ccg.github.io/security-vocab/#cipherAlgorithm)
 
 <h4 id="cipherData"><a href="#cipherData">cipherData</a></h4>
 
-Deprecated, do not use.
+Cipher data an opaque blob of information that is used to specify an encrypted message.
```
Comment (gjgd) [inline security.md]:
```suggestion
Cipher data is an opaque blob of information that is used to specify an encrypted message.
```

File: security.md [hunk 2]
```diff
@@ -66,67 +76,91 @@ Deprecated, do not use.
 
 <h4 id="expiration"><a href="#expiration">expiration</a></h4>
 
-Deprecated, do not use.
+See [vc-data-model](https://www.w3.org/TR/vc-data-model/#expiration).
```
Comment (gjgd) [inline security.md]: The example in this link describes the `expirationDate` property, not `expiration`

File: security.md [hunk 3]
```diff
@@ -66,67 +76,91 @@ Deprecated, do not use.
...
 
-Deprecated, do not use.
+A public key property is used to specify a URL that contains information about a public key.
+
+See [w3c-ccg](https://w3c-ccg.github.io/security-vocab/#publicKey).
 
 <h4 id="publicKeyBase58"><a href="#publicKeyBase58">publicKeyBase58</a></h4>
 
-Deprecated, do not use.
+You should use publicKeyJwk if possible.
```
Comment (gjgd) [inline security.md]: This statement should either be motivated by a list of reasons, or removed

File: security.md (hunk 2 above)
Comment (OR13) [reply security.md]: I think they are synonyms... its hard to tell..

File: security.md (hunk 3 above)
Comment (OR13) [reply security.md]: The reason is that publickKeyJwk is valid for all supported key formats according to the did core spec. I will update.

**Pull request merged**